ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/DefaultSynchronizer.java

```diff
@@ -157,13 +156,10 @@ private void handleFastSyncResult(final FastSyncState result, final Throwable er
       // We've been shutdown which will have triggered the fast sync future to complete
       return;
     }
-    final Throwable rootCause = ExceptionUtils.rootCause(error);
-    if (rootCause instanceof FastSyncException) {
-      LOG.error(
-          "Fast sync failed ({}), switching to full sync.",
-          ((FastSyncException) rootCause).getError());
-    } else if (error != null) {
-      LOG.error("Fast sync failed, switching to full sync.", error);
+    if (error != null) {
+      final Throwable rootCause = ExceptionUtils.rootCause(error);
+      stop();
+      throw new RuntimeException("Fast sync failed.", rootCause);
```

Comment:
Throwing a RuntimeException here doesn't really make sense - it will wind up being ignored because this method is called via:
`fastSyncDownloader.get().start().whenComplete(this::handleFastSyncResult);`

so the future returned by `whenComplete` will now have completed exceptionally but nothing uses that future so it just gets ignored.  You can just return instead of throwing.

However, this leaves the Besu instance running but with no ongoing sync process at all which seems very odd. As time goes on, ongoing log messages from other components will likely make it hard to see that the synchroniser has completely failed.

I'd have thought we'd either want to completely crash (exit the process) or continue retrying fast sync. My preference would be to just keep retrying because exiting the process is pretty brutal. If we do exit we'd need to provide a way for things that embed Besu to disable it or do something else (ie don't just call `System.exit` directly).